lib/hwm/hwm.go

```diff
@@ -120,6 +139,15 @@ func (trie *HighWatermarkTracker) DeleteOldKeys(hwmLimit uint64) []string {
 	return deletedKeys
 }
 
+func (trie *HighWatermarkTracker) ToMap() map[string]uint64 {
```

Comment:
There are two types of tests: you can put tests in the same package `foo`, say or in a special test package in the same directory `foo_test`.  The latter type test using your public API, so more of a black box test.  Both sorts of test are favoured.  In this case, it's borderline but adding a new (perfectly reasonable) public method seemed a lot cleaner than poking into the awkward internals.